src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,85 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        checkSize();
+        final Node<K, V> newNode = new Node<>(key, value);
+        int index = getIndex(key);
+        Node<K, V> currentNode = table[index];
+        while (currentNode != null) {
+            if (currentNode.key == newNode.key || currentNode.key != null
+                    && currentNode.key.equals(newNode.key)) {
+                currentNode.value = newNode.value;
+                break;
+            } else if (currentNode.next == null) {
+                currentNode.next = newNode;
+                size++;
+                break;
+            }
+            currentNode = currentNode.next;
+        }
+        if (currentNode == null) {
+            table[index] = newNode;
+            size++;
+        }
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> node = table[getIndex(key)];
+        while (node != null) {
+            if (node.key == key || node.key != null && node.key.equals(key)) {
+                return node.value;
+            } else {
+                node = node.next;
+            }
```

Comment:
do not use **else** after **return**
```suggestion
            } 
                node = node.next;
```